publish_example_service_test.go

```diff
@@ -0,0 +1,479 @@
+
+	name := get_table_value(table, "name")
+	price := get_table_value(table, "price")
+	endpoint := get_table_value(table, "endpoint")
+	tags := get_table_value(table, "tags")
+	description := get_table_value(table, "description")
+
+	command := ExecCommand{
+		Command:   "snet",
+		Directory: example_service_dir,
+		Input:     []string{"", "", name, "", price, endpoint, tags, description},
+		Args:      []string{"service", "init"},
+	}
+
+	return run_command(command)
+}
+
+func example_service_is_published_to_network(table *gherkin.DataTable) error {
+
+	agent_factory_address := get_table_value(table, "agent factory address")
+	registry_address := get_table_value(table, "registry address")
+
+	args := []string{
+		"service", "publish", "local",
+		"--config", "./service.json",
+		"--agent-factory-at", agent_factory_address,
+		"--registry-at", registry_address,
+	}
+
+	command := ExecCommand{
+		Command:   "snet",
+		Directory: example_service_dir,
+		Input:     []string{"y", "y"},
+		Args:      args,
+	}
+
+	return run_command(command)
+}
+
+func exampleservice_is_run_with_snet_daemon(table *gherkin.DataTable) error {
+
+	daemon_port := get_table_value(table, "daemon port")
+	ethereum_endpoint_port := get_table_value(table, "ethereum endpoint port")
+	passthrough_endpoint_port := get_table_value(table, "passthrough endpoint port")
+
+	agent_contract_address := get_table_value(table, "agent contract address")
+	private_key := get_table_value(table, "private key")
+
+	snetd_config_template := `
+	{
+    "AGENT_CONTRACT_ADDRESS": "%s",
+    "AUTO_SSL_DOMAIN": "",
+    "AUTO_SSL_CACHE_DIR": "",
+    "BLOCKCHAIN_ENABLED": true,
+    "CONFIG_PATH": "",
+    "DAEMON_LISTENING_PORT": %s,
+    "DAEMON_TYPE": "grpc",
+    "DB_PATH": "./db",
+    "ETHEREUM_JSON_RPC_ENDPOINT": "http://localhost:%s",
+    "EXECUTABLE_PATH": "",
+    "LOG_LEVEL": 5,
+    "PASSTHROUGH_ENABLED": true,
+    "PASSTHROUGH_ENDPOINT": "http://localhost:%s",
+    "POLL_SLEEP": "",
+    "PRIVATE_KEY": "%s",
+    "SERVICE_TYPE": "jsonrpc",
+    "SSL_CERT": "",
+    "SSL_KEY": "",
+    "WIRE_ENCODING": "json"
+    }`
+
+	snetd_config := fmt.Sprintf(snetd_config_template,
+		agent_contract_address, daemon_port, ethereum_endpoint_port, passthrough_endpoint_port, private_key)
+
+	file := example_service_dir + "/snetd.config.json"
+	err := write_to_file(file, snetd_config)
+
+	if err != nil {
+		return err
+	}
+
+	link_file(env_singnet_repos+"/snet-daemon/build/snetd-linux-amd64", example_service_dir+"/snetd-linux-amd64")
+
+	output_file = log_path + "/example-service.log"
+	output_contains_strings = []string{}
+
+	command := ExecCommand{
+		Command:    example_service_dir + "/scripts/run-snet-service",
+		Directory:  example_service_dir,
+		OutputFile: output_file,
+	}
+
+	err = run_command_async(command)
+
+	if err != nil {
+		return err
+	}
+
+	_, err = check_with_timeout(check_file_contains_strings)
+
+	if err != nil {
+		return err
+	}
+
+	time.Sleep(2 * time.Second)
```

Comment:
Why it is needed? Usually you cannot rely on such delays.

Reply:
Removing the delay requires adding some logging to the snet-daemon and example service that they are ready. I would prefer to make it as a separate fix.